server/src/main/java/com/genymobile/scrcpy/util/IO.java

```diff
@@ -52,6 +63,25 @@ public static void writeFully(FileDescriptor fd, ByteBuffer from) throws IOExcep
                 position += w;
                 from.position(position);
             }
+        } else {
+            try (FileOutputStream output = new FileOutputStream(fd)) {
```

Comment:
Migrate consumer to `Channel` so we don't allocate `FileOutputStream` and `FileChannelImpl` for each `ByteBuffer`?